.travis.yml

```diff
@@ -32,12 +32,12 @@ install:
         libdumb1-dev
     fi
 script:
-  - >
+  - cd build
+  - |
     if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
-      cmake .
-    fi
-  - >
-    if [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
-      cmake .
+      cmake ..
+      cmake --build . --parallel `nproc`
+    elif [[ "${TRAVIS_OS_NAME}" == "osx" ]]; then
+      cmake -GXcode ..
+      cmake --build .
```

Comment:
Is there a reason to force parallel building on Linux and Xcode on OSX?

Reply:
Both are to replace the `make -j2` in a more future proof (IMHO) manner, since Travis may upgrade their machine, and parallel build saves time.

I'm not sure if CMake on macOS will keep defaulting to GNU Make given the current shift to non-GPL software lately, and since Xcode is required up there so...  Like Ninja, Xcode will also try to parallelize the build process.

Reply:
Changing to `cmake --build` is a good idea since that should work with any generator uses. But in that case, it shouldn't matter what generator CMake defaults to on OSX. If Apple drops GNU Make, CMake shouldn't use it by default on OSX since it would be a poor usability design. You should only need to specify a particular generator if you need that particular one (e.g. for a different compiler version, like newer or older Visual Studio), otherwise it should do the right thing by default.

Reply:
Thanks, that's an helpful advise.  I also removed the generator argument for macOS as requested.